apps/dashboard/src/components/activity/activity-filters.tsx

```diff
@@ -28,6 +51,32 @@ export function ActivityFilters({
   className,
 }: ActivityFilters) {
   const { data: workflowTemplates } = useFetchWorkflows({ limit: 100 });
+  const { organization } = useOrganization();
+  const { subscription } = useFetchSubscription();
+
+  const maxActivityFeedRetentionMs = getFeatureForTierAsNumber(
+    FeatureNameEnum.PLATFORM_ACTIVITY_FEED_RETENTION,
+    subscription?.apiServiceLevel ?? ApiServiceLevelEnum.FREE,
+    true
+  );
+
+  const maxActivityFeedRetentionOptions = DATE_RANGE_OPTIONS.map((option) => {
+    const isLegacyFreeTier =
+      subscription?.apiServiceLevel === ApiServiceLevelEnum.FREE &&
+      organization &&
+      organization.createdAt < new Date('2025-02-28');
```

Comment:
Can we drive this from the API (probably the subscription endpoint) to avoid an extra hardcoding of the date?

Reply:
I think using subscription endpoint would break the endpoint for OS version, since its connected to Stripe, private package etc.

I mean its not ideal but we won't be changing this date and its also matched with the db script so I guess its fine.